packages/core/aqua-to-js/src/index.ts

```diff
@@ -27,17 +27,14 @@ interface TsOutput {
   sources: string;
 }
 
-type LanguageOutput = {
-  js: JsOutput;
-  ts: TsOutput;
-};
+type LanguageOutput = JsOutput | TsOutput;
 
 type NothingToGenerate = null;
 
-export default async function aquaToJs<T extends OutputType>(
+export default async function aquaToJs(
   res: CompilationResult,
-  outputType: T,
-): Promise<LanguageOutput[T] | NothingToGenerate> {
+  outputType: OutputType,
+): Promise<LanguageOutput | NothingToGenerate> {
```

Comment:
why you removed generics here?
we know at compile time that if you pass `outputType: "js"` - then `JsOutput` will be returned
now when you removed this we lost this ability and we have to always check if the return value has `types` property in it or not to satisfy TS even though we are completely sure that if we pass "js" there will be a "types" property in the returned value